### PR TITLE
refactor: isolate index handling to BulkDependentResource interface

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java [hunk 1]
```diff
@@ -22,16 +20,16 @@
...
+  private final boolean bulk = this instanceof BulkDependentResource;
 
   protected Creator<R, P> creator;
   protected Updater<R, P> updater;
   protected BulkDependentResource<R, P> bulkDependentResource;
   private boolean returnEventSource = true;
-
-  protected List<ResourceDiscriminator<R, P>> resourceDiscriminator = new ArrayList<>(1);
+  private ResourceDiscriminator<R, P> resourceDiscriminator;
+  private int currentCount;
```
Comment (csviri) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java:32]: Should we address this global variable in this PR, or  rather should make a separate one?

Comment (metacosm) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java:32]: I would say in a different PR but we can do it here if it makes more sense.

**Review by csviri — APPROVED** [review]
LGTM

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java (hunk 1 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java:29]: sure np

**Pull request merged**